software/dlflib/src/datastream/AbstractStream.hpp

```diff
@@ -2,6 +2,7 @@
 
 #include <Arduino.h>
 #include <string.h>
+#include "freertos/semphr.h"
```

Comment:
nit: use angle bracket include since freertos is a library (as opposed to double quote includes, which are used for includes that are part of this project)

```suggestion
#include <freertos/semphr.h>
```